File: javacode/benchmarkpla/Function.java
/*********************************************************
 *          Author : Sajib Kumar Mitra
 *         Position: M.Sc.(1st Semester)/2008-09
 *       Department: Computer Science and Engineering
 *        Institute: University of Dhaka
 *            Email: sajibmitra.csedu@yahoo.com
 *            Phone: +8801710283239
 *********************************************************/
package benchmarkpla;
import java.util.ArrayList;
public class Function {
    public int id;              //ID of a function
    public int productDensity;  //total number of products
//list of used product to build this function
    public ArrayList<Integer> productsList= new ArrayList<Integer>();
//Initialization of a function
    public Function(int funcId){
        id=funcId;
        productDensity=0;
    }
//Add product to function
    public int addProduct(int productId){
        productsList.add(productId);
        productDensity++;
//Return the position of this product in this function
        return productsList.size()-1;
    }
//return the overall function data
    public ArrayList<Integer> getData(){
         return productsList;
    }
//Total number of product to build function
    public int getSize(){
        return productDensity;
    }    
}

File: python/function.py
class Function:
    def __init__(self, funcId):
        self.id = funcId
        self.productDensity = 0
        self.productsList = []

    def addProduct(self, productId):
        self.productsList.append(productId)
        self.productDensity += 1
        return len(self.productsList) - 1

    def getData(self):
        return self.productsList

    def getSize(self):
        return self.productDensity